parlai/core/torch_generator_agent.py

```diff
@@ -1114,15 +1117,21 @@ def _generate(
         encoder_states = model.reorder_encoder_states(encoder_states, inds)
         incr_state = None
 
+        total_decoder_elapsed_time = 0
+        num_decoder_forwards = 0
         for _ts in range(max_ts):
             if all((b.is_done() for b in beams)):
                 # exit early if possible
                 break
 
+            decoder_start_time = time.time()
             score, incr_state = model.decoder(decoder_input, encoder_states, incr_state)
             # only need the final hidden state to make the word prediction
             score = score[:, -1:, :]
             score = model.output(score)
```

Comment:
we might need to do a flag anyway. this code isn't going to be deterministic. pytorch gpu execution is asynchronous and so you can easily be significantly undercounting the model effort here. I believe we have to use CUDA_LAUNCH_BLOCKING to get accurate measurements.

Reply:
Yes, good point - changed